tests/dragonfly/sentinel_test.py

```diff
@@ -0,0 +1,195 @@
+# Output is expected be of even number of lines where each pair of consecutive lines results in a single key value pair.
+# If new_dict_key is not empty, encountering it in the output will start a new dictionary, this let us return multiple
+# dictionaries, for example in the 'slaves' command, one dictionary for each slave.
+def stdout_as_list_of_dicts(cp: subprocess.CompletedProcess, new_dict_key ="") -> list[dict]:
+    lines = cp.stdout.splitlines()
+    res = []
+    d = None
+    if (new_dict_key == ''):
+        d = dict()
+        res.append(d)
+    for i in range(0, len(lines), 2):
+        if (lines[i]) == new_dict_key: # assumes output never has '' as a key
+            d = dict()
+            res.append(d)
+        d[lines[i]] = lines[i + 1]
+    return res
+
+def wait_for(func, pred, timeout_sec, timeout_msg=""):
+    while not pred(func()):
+        assert timeout_sec > 0, timeout_msg
+        timeout_sec = timeout_sec - 1
+        time.sleep(1)
+
+async def await_for(func, pred, timeout_sec, timeout_msg=""):
+    is_func_async = inspect.iscoroutinefunction(func)
+    done = False
+    while not done:
+        val = None
+        if is_func_async: val = await func()
+        else: val = func()
+        done = pred(val)
+        assert timeout_sec > 0, timeout_msg
+        timeout_sec = timeout_sec - 1
+        await asyncio.sleep(1)
+
+
+class Sentinel:
+    def __init__(self) -> None:
+        self.port = 5555
+        self.image = "bitnami/redis-sentinel:latest"
+        self.container_name = "sentinel_test_py_sentinel"
+        self.default_deployment = "my_deployment"
+        self.initial_master_port = 1111
+
+    def start(self):
+        self.stop() # cleanup potential zombies
+
+        cmd = [
+            "docker", "run",
+            "--network=host",
+            "-d",
+            f"--name={self.container_name}",
+            "-e", "REDIS_MASTER_HOST=localhost",
+            "-e", f"REDIS_MASTER_PORT_NUMBER={self.initial_master_port}",
+            "-e", f"REDIS_MASTER_SET={self.default_deployment}",
+            "-e", f"REDIS_SENTINEL_PORT_NUMBER={self.port}",
+            "-e", "REDIS_SENTINEL_QUORUM=1",
+            "-e", "REDIS_SENTINEL_DOWN_AFTER_MILLISECONDS=3000",
+            "bitnami/redis-sentinel:latest"]
+        assert subprocess.run(cmd).returncode == 0, "docker run failed."
+
+    def stop(self):
+        assert subprocess.run(["docker", "rm", "-f", "-v", f"{self.container_name}"]).returncode == 0, "docker rm failed."
+
+    def run_cmd(self, args, sentinel_cmd=True, capture_output=False, assert_ok=True) -> subprocess.CompletedProcess:
+        run_args = ["redis-cli", "-p", f"{self.port}"]
+        if sentinel_cmd: run_args = run_args + ["sentinel"]
+        run_args = run_args + args
+        cp = subprocess.run(run_args, capture_output=capture_output, text=True)
+        if assert_ok:
+            assert cp.returncode == 0, f"Command failed: {run_args}"
+        return cp
```

Comment:
I'm sure the default client libraries support sentinel commands which would have also saved you from writing `stdout_as_list_of_dicts`

Reply:
From a brief look at the client library API it does't seem to cover all commands I've used here and I think we'd want to be able to query the internal state of the sentinel to better asses the integration.